src/Statement/MysqlStatementBuilder.php

```diff
@@ -117,6 +117,21 @@ protected function buildType(array $options): string
                 return 'JSON';
             case 'enum':
                 return sprintf("ENUM('%s')", implode("','", $options['values']));
+            case 'text':
+                if (array_key_exists('length', $options)) {
+                    $sizes = [
+                        'LONGTEXT' => 4294967295,
+                        'MEDIUMTEXT' => 16777215,
+                        'TEXT' => 65535,
+                        'TINYTEXT' => 255
+                    ];
+                    foreach ($sizes as $name => $length) {
+                        if ($options['length'] >= $length) {
+                            return $name;
+                        }
+                    }
+                }
+                return 'TEXT';
```

Comment:
What should be done if the requested length is not satisfied by any of the inbuilt types? Currently it will fallback to `TEXT` but perhaps we should use the largest available type or throw an exception.

Reply:
Good point. I'd probably vote for an exception otherwise we are going to create columns with a different length than the user expected. Let's educate the user rather than be too permissive.